cli/generate/src/prepare_grammar/expand_tokens.rs

```diff
@@ -114,8 +95,25 @@ impl NfaBuilder {
     fn expand_rule(&mut self, rule: &Rule, mut next_state_id: u32) -> Result<bool> {
         match rule {
             Rule::Pattern(s, f) => {
-                let ast = parse::Parser::new().parse(s)?;
-                self.expand_regex(&ast, next_state_id, f.contains('i'))
+                // With unicode enabled, `\w`, `\s` and `\d` expand to character sets that are much
+                // larger than intended, so we replace them with the actual
+                // character sets they should represent. If the full unicode range
+                // of `\w`, `\s` or `\d` are needed then `\p{L}`, `\p{Z}` and `\p{N}` should be
+                // used.
+                let s = s
+                    .replace(r"\w", r"[0-9A-Za-z_]")
+                    .replace(r"\s", r"[\t-\r ]")
```

Comment:
This does not match [JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes#types). It should be:
`[\f\n\r\t\v\u0020\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]`

Reply:
It never matched JS in the first place.